app/handlers/base.py

```diff
@@ -305,6 +325,25 @@ def error(self, message, data={}, status=400, extra={}):
         self.set_status(status)
         self.write({"status": "error", "message": message, "data": data, **extra})
 
+        if self.current_user is None:
+            return
+        with DBSession() as session:
+            session.merge(self.current_user)
+            if hasattr(self.current_user, "created_by_id"):
+                user_id = int(self.current_user.created_by_id)
+            else:
+                user_id = int(self.current_user.id)
+
+            api_call = APICall(
+                user_id=user_id,
+                method=self.request.method,
+                uri=self.request.uri.split("?")[0],
```

Comment:
I'm not sure, but maybe worth while to save the parameters as well? Maybe limit the number of characters, just in case we get super-long parameter strings, but those parameters could contain lots of useful info. 

Also, consider clipping the beginning of the string (I don't know if URI contains the full address or just the thing after `api/`)